apps/passwd.c

```diff
@@ -544,6 +548,9 @@ static char *shacrypt(const char *passwd, const char *magic, const char *salt)
     }
 
     if (strncmp(salt, rounds_prefix, sizeof(rounds_prefix) - 1) == 0) {
+        if (rounds != 0) /* rounds specified explicitly and in salt */
+            return NULL;
```

Comment:
Please output an appropriate error message here. Also move it after declarations.